### PR TITLE
Updated wallet.sol

### PR DESCRIPTION
**Pull request by frozeman**
This PR introduces 4 changes:

- The operation hash includes the block number, so that every operation hash is unique.
This is necessary for any interface to display send requests to other wallet owners. As the previous operation hash could be the same when the recipient and the value is the same, it is impossible to determine whether the operation is a new one or an old, which was already confirmed by all owners at a previous time.
- add function `hasConfirmed(operation, owner)`, which allows to check if a owner has already confirmed a transaction.
- moved the variables to the top of the contracts, for better visibility.
- Add identifier to the created event, to be able to track the creation of the contract and receive the contract address though this log. See the following pattern: https://gist.github.com/frozeman/655a9325a93ac198416e

### CODE REVIEW

File: wallet/wallet.sol [hunk 1]
```diff
@@ -264,7 +291,7 @@ contract Wallet is multisig, multiowned, daylimit {
     // If not, goes into multisig process. We provide a hash on return to allow the sender to provide
     // shortcuts for the other confirmations (allowing them to avoid replicating the _to, _value
     // and _data arguments). They still get the option of using them if they want, anyways.
-    function execute(address _to, uint _value, bytes _data) external onlyowner returns (bytes32 _r) {
+    function execute(address _to, uint _value, bytes _data) external returns (bytes32 _r) {
```
Comment (chriseth) [inline wallet/wallet.sol]: Did you remove `onlyowner` on purpose here?

Comment (frozeman) [inline wallet/wallet.sol]: hm. probably.

**Comment by gavofyork** [comment]
please revert the stylistic changes ("moved the variables to the top of the contracts, for better visibility.") so that the functional changes are visible.

**Comment by gavofyork** [comment]
gah. that didn't actually merge it; please resubmit rebased without the stylistic stuff and i'll actually merge.

**Comment by frozeman** [comment]
ok

**Comment by gavofyork** [comment]
hmm... actually, maybe it did...

**Comment by frozeman** [comment]
haha, so we have nice variables on the top : )

**Comment by frozeman** [comment]
Btw could we rename this repo to `smart-contracts`, as `dapp-bin` is a bit misleading now.

**Comment by frozeman** [comment]
If you rename this  repo, it `dapp-bin` would still redirect to the new repo name, as long as we don't create a new repo called this way. Like https://github.com/ethereum/ethereum.js still works. So nothing gets lost.

**Pull request merged**